File: Funtion/Grayscale3DPlotGenerator.m
% 读取图像
image = imread('path/to/your/image.jpg');

% 获取图像的宽和高
[height, width] = size(image);

% 生成坐标网格
[x, y] = meshgrid(1:width, 1:height);

% 创建3D图
figure;
surf(x, y, double(image), 'EdgeColor', 'none');

% 设置轴标签
xlabel('Width');
ylabel('Height');
zlabel('Grayscale Value');

% 显示图像
title('Grayscale Distribution');

File: Funtion/Grayscale3DPlotGenerator.py
import numpy as np
import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import Axes3D
import cv2

def generate_3d_grayscale_distribution(image_path):
    # 读取图像
    image = cv2.imread(image_path, cv2.IMREAD_GRAYSCALE)

    # 获取图像尺寸
    height, width = image.shape

    # 计算灰度值分布
    grayscale_distribution = np.zeros((256,))

    for i in range(height):
        for j in range(width):
            grayscale_value = image[i, j]
            grayscale_distribution[grayscale_value] += 1

    # 生成3D灰度分布图
    x = np.arange(256)
    y = np.arange(height)
    x, y = np.meshgrid(x, y)

    fig = plt.figure()
    ax = fig.add_subplot(111, projection='3d')
    ax.bar3d(x.ravel(), y.ravel(), 0, 1, 1, grayscale_distribution)

    ax.set_xlabel('灰度值')
    ax.set_ylabel('图像高度')
    ax.set_zlabel('像素数量')

    plt.show()

# 替换为你自己的图像路径
image_path = 'path/to/your/image.jpg'
generate_3d_grayscale_distribution(image_path)
